beacon-chain/monitor/service.go

```diff
@@ -59,6 +84,183 @@ type Service struct {
 	lastSyncedEpoch             types.Epoch
 }
 
+// NewService sets up a new validator monitor instance when given a list of
+// validator indices to track
+func NewService(ctx context.Context, config *ValidatorMonitorConfig, tracked []types.ValidatorIndex) (*Service, error) {
+	ctx, cancel := context.WithCancel(ctx)
+	r := &Service{
+		config:                      config,
+		ctx:                         ctx,
+		cancel:                      cancel,
+		TrackedValidators:           make(map[types.ValidatorIndex]interface{}, len(tracked)),
+		latestPerformance:           make(map[types.ValidatorIndex]ValidatorLatestPerformance),
+		aggregatedPerformance:       make(map[types.ValidatorIndex]ValidatorAggregatedPerformance),
+		trackedSyncCommitteeIndices: make(map[types.ValidatorIndex][]types.CommitteeIndex),
+	}
+	for _, idx := range tracked {
+		r.TrackedValidators[idx] = nil
```

Comment:
This map is not filled with real data in this PR. I presume it's done somewhere else?

Reply:
The fact that the key is there is used to check that the `idx` is tracked.

Reply:
Instead of assigning nil, why don't we assign the actual struct

Reply:
You mean one of the performance structures? because those take a long time to be available, this would block service start until we are fully synced at the minimum.